org-roam-capture.el

```diff
@@ -279,23 +280,27 @@ This function is used solely in Org-roam's capture templates: see
 
 (defun org-roam-capture--convert-template (template)
   "Convert TEMPLATE from Org-roam syntax to `org-capture-templates' syntax."
-  (let* ((copy (copy-tree template))
-         converted
-         org-roam-plist
-         key
-         val)
-    ;;put positional args on converted template
-    (dotimes (_ 5)
-      (push (pop copy) converted))
-    (while (setq key (pop copy)
-                 val (pop copy))
-      (if (member key org-roam-capture--template-keywords)
-          (progn
-            (push val org-roam-plist)
-            (push key org-roam-plist))
-        (push key converted)
-        (push val converted)))
-    (append (nreverse converted) `(:org-roam ,org-roam-plist))))
+  (pcase template
+    (`(,(pred stringp) ,(pred stringp))
+     template)
+    (_
+     (let ((copy (copy-tree template))
+           converted
+           org-roam-plist
+           key
+           val)
+       ;; Convert positional args before taking care of the plist-args
+       (dotimes (_ 5)
+         (push (pop copy) converted))
+       (while (setq key (pop copy)
+                    val (pop copy))
+         (if (member key org-roam-capture--template-keywords)
+             (progn
+               (push val org-roam-plist)
+               (push key org-roam-plist))
+           (push key converted)
+           (push val converted)))
+       (append (nreverse converted) `(:org-roam ,org-roam-plist))))))
```

Comment:
```suggestion
  (pcase template
    (`(,key ,description) template)
    (`(,key ,description ,type ,target . ,rest)
     (let ((converted `(,key ,description ,type ,target
                             ,(unless (keywordp (car rest)) (pop rest))))
           org-roam-plist
           key
           val)
       (while (setq key (pop rest)
                    val (pop rest))
         (if (member key org-roam-capture--template-keywords)
             (progn (push val org-roam-plist)
                    (push key org-roam-plist))
           (push key converted)
           (push val converted)))
       (append converted `(:org-roam ,org-roam-plist))))
    (_ (user-error "Invalid capture template format: %s" template))))
```
A pcase is a good choice for a structure like an org-capture-template.
I wouldn't worry too much about type checking every element of the template, though, because beyond simple checks it's enough work to become its own package (that's a large part of what doct does).
Note in the second case the use of the dotted notation to capture the rest of the list. It's a useful pattern.

Reply:
Actually, I neglected to consider the order of the options when appending that, one second...

Reply:
> Note in the second case the use of the dotted notation to capture the rest of the list. It's a useful pattern.

Aw, I knew there was a way to do this!  Thanks a bunch for this, although I wish I could have found it on my end, especially since I'm supposed to know how lists work in a Lisp. \*sigh\*

Let me review your code and implement it.

Reply:
Your solution is much more elegant than mine, and you're making a better use of `pcase`.  It's always a pleasure to study your code and learn new ways of using it.  Thanks!

Reply:
Applied in 7107ce4efa658ba44f0294c914c9d4b7b4df5039.